File: tests/fixtures/edgecase_repo/ruby/module6.rb
require_relative "./module5"

module Example
  class Base6
    def initialize(name)
      @name = name
    end
  end

  class Service6 < Base6
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_6(value)
  value
end

File: tests/fixtures/edgecase_repo/php/module6.php
<?php
namespace Example\Gen;

use Example\Gen\Base6;
use Example\Gen\Handler6 as H6;

class Base6
{
    protected $name;
}

interface Handler6
{
    public function handle($payload);
}

class Service6 extends Base6 implements Handler6
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_6($value) {
    return $value;
}
